posts/2023-09-30-keycloak-with-openliberty.adoc

```diff
@@ -0,0 +1,126 @@
+---
+layout: post
+title: "Protect your applications with Jakarta Security, MicroProfile JWT, and Keycloak"
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/jimmy1wu
+author_github: https://github.com/jimmy1wu
+seo-title: Protect your applications with Jakarta Security, MicroProfile JWT, and Keycloak - OpenLiberty.io
+seo-description: Open Liberty can easily integrate with Keycloak, which can be used as an OpenID Connect Provider, using Jakarta Security and MicroProfile JWT.
+blog_description: "Open Liberty can easily integrate with Keycloak, which can be used as an OpenID Connect Provider, using Jakarta Security and MicroProfile JWT."
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= Protect your applications with Jakarta Security, MicroProfile JWT, and Keycloak
+Jimmy Wu <https://github.com/jimmy1wu>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+
+Open Liberty can easily integrate with Keycloak, which can be used as an OpenID Connect Provider, using Jakarta Security and MicroProfile JWT.
+In this blog post, a Keycloak server has been setup with a realm called `openliberty` which contains a client called `sample-openliberty-keycloak` and the realm roles of `admin` and `user`. Additionally in the `sample-openliberty-keycloak` client, client authentication has been enabled, `http://localhost:9090/Callback` has been added as a valid redirect URI, and the `microprofile-jwt` client scope has been set to `Default`.
+
+== Obtaining an access token from Keycloak using Jakarta Security
+
+With the new `@OpenIdAuthenticationMechanismDefinition` annotation introduced in link:https://openliberty.io/docs/latest/reference/feature/appSecurity-5.0.html[Jakarta Security 3.0], you can easily authenticate users of your application with Keycloak and obtain an access token.
```

Comment:
Similarly here, may be nice to say what users would have had to have done before this new annotation (i.e. I assume a longer or harder process) and then say why this new annotation and keycloak make dev lives easier. Want more why here as well as technical how.